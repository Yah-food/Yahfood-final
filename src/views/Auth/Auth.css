
  
h4 {
    margin: 10px;
    cursor: pointer;
    padding-top: 40px;
  }
  
h4.active {
    border-bottom: 2px solid black;
  }