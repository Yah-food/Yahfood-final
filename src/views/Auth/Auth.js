import LoginForm from '../../components/LoginForm/LoginForm';
import { useState } from 'react';
import { signInUser, signUpUser } from '../../services/users';
import classNames from 'classnames';
import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';

export default function Auth({ setCurrentUser }) {
  const [type, setType] = useState('signin');
  const [errorMessage, setErrorMessage] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const resp =
        type === 'signin' ? await signInUser(email, password) : await signUpUser(email, password);
      history.push('/profile');
      setCurrentUser(resp);
    } catch {
      setErrorMessage('error. invalid login.');
    }
  };
  return (
    <div>
      <div className="tabs flex justify-center">
        <h1
          onClick={() => {
            setType('signin');
          }}
          className={classNames({ active: type === 'signin' })}
        >
          Sign In
        </h1>
        <h1
          onClick={() => {
            setType('signup');
          }}
          className={classNames({ active: type === 'signup' })}
        >
          Sign Up
        </h1>
      </div>

      <h1>Type: {type}</h1>
      <p>{errorMessage}</p>
      <LoginForm
        errorMessage={errorMessage}
        email={email}
        setEmail={setEmail}
        password={password}
        setPassword={setPassword}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
