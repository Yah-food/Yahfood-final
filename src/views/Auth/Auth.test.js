import { render } from '@testing-library/react';
import Auth from './Auth';

test('renders login form page', () => {
  const container = render(<Auth />);
  expect(container).toMatchSnapshot();
});
