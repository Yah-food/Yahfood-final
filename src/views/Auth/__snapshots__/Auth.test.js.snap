// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login form page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="tabs"
        >
          <h1
            class="active"
          >
            Sign In
          </h1>
          <h1
            class=""
          >
            Sign Up
          </h1>
        </div>
        <h1>
          Type: 
          signin
        </h1>
        <p />
        <div>
          <form>
            <p />
            <label>
              Email:
              <input
                class="border-black border-solid border-2"
                type="email"
                value=""
              />
            </label>
            <label>
              Password:
              <input
                class="border-black border-solid border-2"
                type="password"
                value=""
              />
            </label>
            <input
              class="border-black border-solid border-2"
              type="submit"
            />
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="tabs"
      >
        <h1
          class="active"
        >
          Sign In
        </h1>
        <h1
          class=""
        >
          Sign Up
        </h1>
      </div>
      <h1>
        Type: 
        signin
      </h1>
      <p />
      <div>
        <form>
          <p />
          <label>
            Email:
            <input
              class="border-black border-solid border-2"
              type="email"
              value=""
            />
          </label>
          <label>
            Password:
            <input
              class="border-black border-solid border-2"
              type="password"
              value=""
            />
          </label>
          <input
            class="border-black border-solid border-2"
            type="submit"
          />
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
